ocaml/xapi/xapi_vbd_helpers.ml

```diff
@@ -127,17 +127,29 @@ let valid_operations ~expensive_sharing_checks ~__context record _ref' : table =
     end
   ) vm_current_ops;
 
-  (* HVM guests only support plug/unplug IF they have PV drivers *)
+  (* HVM guests MAY support plug/unplug IF they have PV drivers. Assume
+   * all drivers have such support unless they specify that they do not. *)
   (* They can only eject/insert CDs not plug/unplug *)
   let vm_gm = Db.VM.get_guest_metrics ~__context ~self:vm in
   let vm_gmr = try Some (Db.VM_guest_metrics.get_record_internal ~__context ~self:vm_gm) with _ -> None in  
   if power_state = `Running && Helpers.has_booted_hvm ~__context ~self:vm then begin
-    (match Xapi_pv_driver_version.make_error_opt (Xapi_pv_driver_version.of_guest_metrics vm_gmr) vm with
-     | Some(code, params) -> set_errors code params [ `plug; `unplug; `unplug_force ]
-     | None -> ());
+    let plug_ops = [ `plug; `unplug; `unplug_force ] in
```

Comment:
I'll admit I didn't look into that, and I don't know the thinking that led to the decision to do it that way long ago. I just kept the behaviour the same as before except basing the decision on the new field rather than just the PV driver version. If it's not what we want, then I suggest that changing it should probably be a separate pull-request.